hathor/transaction/base_transaction.py

```diff
@@ -174,7 +195,7 @@ def __init__(self,
         """
         self.nonce = nonce
         self.timestamp = timestamp or int(time.time())
-        self.version = version
+        self.signal_bits, self.version = TxVersion.extract_from_bytes(version)
```

Comment:
And instead of using a "version" with possibly 2 bytes here, we'd use a "normal" version with no signal bits included, and add a `signal_bits` argument to `__init__` arguments.

Reply:
I did it like this to more or less keep the same interface we have now, and to keep the historical fact that this extra byte is part of the version: for users of this class, nothing is changed. For example, in the `hathor/client.py` code, it could be that `data['version']` is being sent with two bytes. And I'm not too sure that there aren't other parts of the code that could be relying on the fact that `version` occupies two bytes.

If you think there's no risk is doing it like you suggested, I agree with you that it would be better to have `signal_bits` completely separated from `version` and `TxVersion`. What does @msbrogli think, too?

Reply:
I think that the internal representation may be not related to the serialization/deserialization. I mean, we can look at `bytes(tx)` as having a 2-byte `version` but also as having a 1-byte version and a 1-byte `signal_bits`. In this sense, I feel it makes sense to split it in another parameter.

Even if the version really has two bytes (e.g., we might use only 4 bits of the second byte for bit signaling), I feel that they are different animals that are serialized together for historical reasons.

Reply:
Ok, makes sense. I'll implement @jansegre's suggestion then

Reply:
Done in https://github.com/HathorNetwork/hathor-core/pull/623/commits/64baf0c73c0f872c060bed69ac4f5f624bb9ea8d